NRPT-163: Updates Core importer mapping

KitArmstrong opened the pull request:
Ticket: https://bcmines.atlassian.net/browse/NRPT-163

Updates the Core importer to map the status and commodities properly.

NickPhura commented on api/src/integrations/core/mine-utils.js:
```diff
@@ -101,8 +103,13 @@ class Mines extends BaseRecordUtils {
 
-    const latestStatus = mineRecord.mine_status.pop();
-    return (latestStatus && latestStatus.status_labels.join(' ')) || '';
+    if (!mineRecord.mine_status.length) {
```
Do we need a null check for mine_status? Or should it always be non-null?

KitArmstrong replied:
Updated

KitArmstrong commented on api/src/integrations/core/mine-utils.js:
```diff
@@ -78,13 +78,17 @@ class Mines extends BaseRecordUtils {
-      commodityTypes.forEach(commodity => {
-        if (commodity.mine_tenure_type_codes.includes(type.mine_tenure_type_code)) {
-          commodities.push(commodity.description);
+    for (const mineType of mineRecord.mine_type) {
```
`mine_type` is checked above so that a proper error is thrown if it is missing.

KitArmstrong commented:
I fixed a unit test and added in a few additional null checks. The data coming from Core API has null values, but the properties on the objects always exist. I still added the checks in case that ever changes though.
